app/api/chemotion/search_api.rb

```diff
@@ -522,12 +522,11 @@ def elements_by_scope(scope, collection_id = @c_id)
         end
 
         post do
-          Usecases::Search::StructureSearch.new(
-            collection_id: @c_id,
-            params: params,
-            user: current_user,
-            detail_levels: @dl,
-          ).perform!
+          samples = sample_structure_search
```

Comment:
Sure that all logic that used to be in `StructureSearch.perform!` is performed  here? What about the special treatment of cell lines that used to be in `StructureSearch.perform!`, i.e., 

```Ruby
results = @shared_methods.serialization_by_elements_and_page(@elements, '')
results['cell_lines'] = { elements: [], ids: [], page: 1, perPage: params['per_page'], pages: 0, totalElements: 0, error: '' }
results
```

Reply:
Debug mode says:
result ['cell_lines'] => {"elements"=>[], "totalElements"=>0, "page"=>1, "pages"=>0, "perPage"=>15, "ids"=>[]}
So the serialization_by_elements_and_page of the search_api already sets this value. Maybe this isn't set in shared_methods.serialization_by_elements_and_page.

Reply:
Thanks for the clarification. On closer inspection, `cell_lines` wasn't includes in `@elements` in `StructureSearch`. That's why the line `results['cell_lines'] = { elements: [], ids: [], page: 1, perPage: params['per_page'], pages: 0, totalElements: 0, error: '' }` was added. Conclusion: it's not necessary to replicate the special treatment of `cell_lines`.